validation-test/compiler_crashers_2_fixed/sr11236.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: not %target-swift-frontend -typecheck %s
+
+public func trichotomy<T : Comparable>(x: T, y: T) -> some Comparable {
+  if x < y { return -1 } 
+  else if x == y { return 0 }
+  return 1
+}
+
+public func myTri<T: Comparable, U: Comparable> (retval: UnsafeMutablePointer<U>, x: UnsafeMutablePointer<T>, y: UnsafeMutablePointer<T>) {
+  retval.initialize(to: trichotomy(x: x, y: y))
+}
+
+print("Hello, World!")
```

Comment:
Since the crash is located in the body of `myTri` this `print` is not really necessary.